[Clifton Felix] iP

cliftonfelix opened the pull request:
### **Tob Tob**

> "Productivity is never an accident. It is always the result of a commitment to excellence, intelligent planning, and focused effort." - Paul J. Meyer ([source](https://www.brainyquote.com/quotes/paul_j_meyer_393225))

Your brain is too precious to remember things you need to do. 

Why would you use your brain when you can use Tob Tob's brain for those unnecessary things?

Use Tob Tob now! It's

- **FREE**
- text-based
- **SUPER EASY** to use
- **Faster** than _Lightning McQueen_

Now, put your mind in peace! All you need to do is,

1. Download it from [here](https://github.com/cliftonfelix/ip/releases).
2. Double-click it.
3. Migrate the tasks from your brain to Tob Tob's brain.
4. Let it manage your tasks for you 😉

Features:


- [x]  Managing tasks
- [x]  Managing deadlines
- [x]  Finding specific task names
- [ ]  Reminders (coming soon)

Commit:
Add Gradle support

cliftonfelix committed:
Level-1: Greet, Echo, Exit

cliftonfelix committed:
moving some variables to the top

cliftonfelix committed:
Level-2: Add, List

cliftonfelix committed:
Level-3: Mark as Done

cliftonfelix committed:
ignore java class files

cliftonfelix committed:
Level-4: ToDo, Event, Deadline

cliftonfelix committed:
A-TextUiTesting

cliftonfelix committed:
Level-5: Handle Errors

cliftonfelix committed:
Level-6: Delete

cliftonfelix committed:
change code structure

cliftonfelix committed:
include more test cases

cliftonfelix committed:
filled in expected.txt

cliftonfelix committed:
Level-7: Save

cliftonfelix committed:
Add notes.txt to .gitignore

cliftonfelix committed:
Level-8: Dates and Times

cliftonfelix committed:
Solved conflict

cliftonfelix committed:
A-MoreOOP

cliftonfelix committed:
Add text-ui-test to .gitignore

cliftonfelix committed:
Fix error in .gitignore

cliftonfelix committed:
A-Packages

cliftonfelix committed:
Add data/duke.txt

cliftonfelix committed:
Merge branch 'add-gradle-support' of github.com:nus-cs2103-AY2223S1/ip

cliftonfelix committed:
A-Gradle

cliftonfelix committed:
Fix java import order

cliftonfelix committed:
A-JUnit

cliftonfelix committed:
A-Jar

cliftonfelix committed:
A-JavaDoc

cliftonfelix committed:
Add javadoc folder in .gitignore

cliftonfelix committed:
Add A-CodingStandard

cliftonfelix committed:
Add Level-9: Find

cliftonfelix committed:
Merge branch-Level-9 to master

cliftonfelix committed:
Fix JavaDoc

cliftonfelix committed:
Remove unnecessary import statements

cliftonfelix committed:
Fix Executor JavaDoc checkstyle violation

cliftonfelix committed:
Add params to DukeException JavaDoc

cliftonfelix committed:
Restructure code

cliftonfelix committed:
Add error for bye command with commandDescription

deeyonn commented on build.gradle:
```diff
@@ -0,0 +1,67 @@
+    enableAssertions = true
+}
+
+checkstyle {
+    toolVersion = '10.2'
+}
```
This change will stop checkstyle from checking your JUnit tests in `src\test`
```suggestion
checkstyle {
    toolVersion = '10.2'
    sourceSets  = [project.sourceSets.main]
}
```

deeyonn commented on src/main/java/belly/Belly.java:
```diff
@@ -0,0 +1,91 @@
+ * Represents a storage system to load data from or save data to local hard disk.
+ */
+public class Belly {
+    public static final String FILE_DIRECTORY = "../data/";
```
Does this work on all platforms Windows, Linux, etc?
You can try using the relative home directory.
```java
public Belly(String folderName) throws DukeException {
    String userHome = System.getProperty("user.home");
    this.storagePath = Paths.get(userHome, ".duke", "belly", folderName);
    try {
        Files.createDirectories(storagePath);
    } catch (UnsupportedOperationException | SecurityException | IOException e) {
        throw new DukeException(e.getMessage());
    }
}
```

deeyonn reviewed (commented):
Looks great to me! I really like the creativity of using names like Brain and Belly. 😄

deepimpactmir commented:
Hi! Code looks clean to me! I particularly liked how you did your DukeException class. I'm also intrigued by your way of having a Executor class to handle the commands instead of having the commands themselves implement the execute method.

deepimpactmir reviewed (commented):
Hi! Code looks clean to me! I particularly liked how you did your DukeException class. I'm also intrigued by your way of having a Executor class to handle the commands instead of having the commands themselves implement the execute method.

cliftonfelix committed:
Add Level-10 functionality

cliftonfelix committed:
Add error message for unrecognized file format

cliftonfelix committed:
Fix NullPointerException

cliftonfelix committed:
Add javadoc and fix checkstyle issues

cliftonfelix committed:
Enhance GUI look

cliftonfelix committed:
Fix GUI Background

cliftonfelix committed:
Fix bug when user clears tasks

After user delete all tasks and rerun the app,
Tob Tob shows File Modified error

It shouldn't show the error

Let's delete the file content if the user delete all tasks.

File content is deleted instead of writing an empty string to the file
because writing empty string resulted in a file having an empty line
and scanner will still read the file.

mlzt2000 commented on src/main/java/task/Deadline.java:
```diff
@@ -0,0 +1,54 @@
+     */
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + datetime.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
```
For this sort of more complex String concatenation, would using String.format() be clearer?

mlzt2000 commented on src/main/java/tobtob/TobTobException.java:
```diff
@@ -0,0 +1,129 @@
+ * Represents an error thrown by TobTob
+ */
+public class TobTobException extends Exception {
+    private static final String EMPTY_COMMAND_DESCRIPTION_ERROR_MESSAGE =
```
I like the use of constants to use as error messages here.

mlzt2000 commented on src/main/java/brain/Brain.java:
```diff
@@ -0,0 +1,134 @@
+     *
+     * @return {@code int}
+     */
+    public int size() {
```
Should this be a getter instead, and hence names getSize()?

cliftonfelix committed:
Add assertions to Brain and Executor

There is no assertions used in the code

To make sure that the program state is not violated

Let's add Assertions to the code

cliftonfelix committed:
Fix Code Quality

cliftonfelix committed:
Merge pull request #1 from cliftonfelix/branch-A-Assertions

Add assertions to Brain and Executor

cliftonfelix committed:
Merge pull request #2 from cliftonfelix/branch-A-CodeQuality

Fix Code Quality

cliftonfelix committed:
Add BCD-Extension (C-Help)

cliftonfelix committed:
Merge branch 'master' of github.com:cliftonfelix/ip into branch-BCD-Extension

cliftonfelix committed:
Fix getSize in assert

cliftonfelix committed:
Fix TextField to only clear when user starts typing

cliftonfelix committed:
Add User Guide

cliftonfelix committed:
Add Ui.png

cliftonfelix committed:
Rearrange help message

cliftonfelix committed:
Update README.md

cliftonfelix committed:
Change font to a universal font
